api/envoy/config/core/v3/base.proto

```diff
@@ -160,7 +162,11 @@ message Node {
   // :ref:`CDS <config_cluster_manager_cds>`, and :ref:`HTTP tracing
   // <arch_overview_tracing>`, either in this message or via
   // :option:`--service-cluster`.
-  string cluster = 2;
+  string cluster = 2 [(udpa.annotations.field_migrate).oneof_promotion = "cluster_specifier"];
+
+  // [#not-implemented-hide:]
+  udpa.core.v1.ResourceLocator cluster_resource_locator = 12
+      [(udpa.annotations.field_migrate).oneof_promotion = "cluster_specifier"];
```

Comment:
I haven't been tracking the back and forth, but I skimmed this PR and let my try to summarize what I think @htuch concerns are and some of my own thoughts:
1) I think @htuch thinks this ResourceLocator can potentially point to multiple targets, which is very different than what we have today. @htuch is that right? If that is right, is it up to the impl to fail to load if that is the case in a location where that doesn't make sense.
2) There is some discussion about a flat string vs. a structured impl, right? Is the summary there that we want the structured impl in the API even though we might store a flat string internally?

My thoughts:
1) I'm concerned it's going to be very painful to keep track of all cases of this. Even though it would also be painful, should we deprecate existing string clusters and replace with only ResourceLocator or a message that wraps a oneof between a flat string and a ResourecLocator?
2) Are we sure that every change in this PR makes sense? Concretely, does the changing the Node field (this field) actually make sense? This is an opaque string, it's not an xDS fetch. Are there are other cases like this that might need discussion?